crates/meilisearch-types/src/tasks.rs

```diff
@@ -360,6 +369,7 @@ impl From<&KindWithContent> for Option<Details> {
             }),
             KindWithContent::DumpCreation { .. } => Some(Details::Dump { dump_uid: None }),
             KindWithContent::SnapshotCreation => None,
+            KindWithContent::UpgradeDatabase { .. } => None,
```

Comment:
```suggestion
            KindWithContent::UpgradeDatabase { from: .. } => Some(Details::UpgradeDatabase { upgrade_from: Some()}),
```

Reply:
There was other issues in the end, it's fixed and tested :ok_hand: